Library/Homebrew/utils/ruby.sh

```diff
@@ -45,15 +57,34 @@ EOS
       then
         HOMEBREW_RUBY_PATH="/System/Library/Frameworks/Ruby.framework/Versions/Current/usr/bin/ruby"
       else
-        HOMEBREW_RUBY_PATH="$(type -P ruby)"
+        HOMEBREW_RUBY_PATH=$(type -P ruby)
```

Comment:
This could have spaces in it so keep the quotes here (and elsewhere)

Reply:
when we capture output of a command with `$( ... )`, quotes are not necessary
```sh
a=$(echo 1 2)
echo $a
1 2
```
Just like they're not necessary around variables inside `[[ ... ]]` unless we're dealing with complex regex-es.

Reply:
Quotes are necessary.
```console
$ echo 'Hello, world!' >'a b
$ cat $(ls)
cat: a: No such file or directory
cat: b: No such file or directory
$ cat "$(ls)"
Hello, world!
```

Reply:
not when you assign to `$( ... )`. Your example is flawed: first call to `cat` uses output of $(ls)` as an input to cat, so `cat` sees two "inputs" -- `a` and `b`. Your second call to `cat` sends output of `$(ls)` as a single argument -- so, cat finds file called `a b` and proceeds as necessary

```
$ cd $(mktemp -d)
$ echo 'Hello, world!' > 'a b'
$ files=$(ls)
```

Reply:
@sjackman shell works in mysterious ways :mage: 
i didn't know that trick before, thanks @maxim-belkin

Reply:
Huh, TIL. Thanks, Maxim.